src/components/atoms/markdown/CodeFence.tsx

```diff
@@ -45,14 +44,27 @@ const CodeFence = (props: React.HTMLProps<HTMLPreElement> = {}) => {
     return (
       <CodeFenceContainer>
         <pre {...otherProps} />
-        <CodeFenceCopyButton
+        <CodeFenceButton
           onClick={() => {
             copy(rawContent)
           }}
           title='Copy to Clipboard'
         >
           <Icon path={mdiContentCopy} />
-        </CodeFenceCopyButton>
+        </CodeFenceButton>
+        {rawContent && <CodeFenceButton
+          onClick={() => {
+            var blob = new Blob([rawContent], { type: "text/plain;charset=utf-8" });            
+            var ext = "";
+            if(props.className!.includes('language-'))
+              ext="."+/language-(.+?)$/.exec(props.className!)![1]
+            saveAs(blob, "snippet" + ext);
```

Comment:
Please use `downloadBlob` method in `lib/download` rather than using `file-saver`.

Reply:
As you wish.